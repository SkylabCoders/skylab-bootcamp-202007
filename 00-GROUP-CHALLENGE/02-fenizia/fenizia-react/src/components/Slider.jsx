import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import productStore from "../stores/productStore";
import { loadProducts } from "../actions/productActions";
import "./Slider.scss";

function Slider({ order, url, text }) {
  const [products, setProducts] = useState(productStore.getProduct(order));

  useEffect(() => {
    productStore.addChangeListener(onChange);
    if (products.length === 0) loadProducts();
    return () => productStore.removeChangeListener(onChange);
  }, [products.length]);

  function onChange() {
    setProducts(productStore.getProduct(order));
  }

  // slice must be done in the store!
  const mainProducts = products.slice(0, 10);
  const currency = "euros";

  return (
    <section className="slider">
      <div className="slider__text">
        <Link className="slider__title" to={url}>
          LIBROS - {text}
        </Link>
        <Link className="slider__view-all" to={url}>
          <p>VER TODO</p>
        </Link>
      </div>
      <div className="slider__content">
        {mainProducts.map((element) => (
          <div className="slider__content-wrapper">
            <div key={element.id} className="slider__item">
              <img
                key={element.id}
                alt={element.product.title}
                className="slider__image"
                // src= {require('../img/el-mentiroso.jpg')}
                src={element.product.cover}
              />
              <div key={element.id} className="slider__item-text">
                <div key={element.id}>{element.product.title}</div>
                <div key={element.id}>{element.product.author}</div>
                <div key={element.id}>
                  {element.product.price} {currency}
                </div>
              </div>
            </div>
          </div>
        ))}
      </div>
    </section>
  );
}

export default Slider;
