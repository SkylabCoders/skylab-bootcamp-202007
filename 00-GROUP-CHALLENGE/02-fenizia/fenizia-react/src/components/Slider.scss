$backColor: white;

.slider {
  height: 80vw;
  background-color: $backColor;
  position: relative;
}

.slider__text {
  height: 2em;
  border: 2px yellow;
  background-color: #FFCC00;
  color: white;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-end;
}

.slider__title {
  margin: auto 0 auto 2rem;
  color: #fff;
  text-decoration-line: none;
  
}

/* .slider__title:after {
  display: inline-block;
  content: ">";
  position: relative;
  margin-left: 4px;
} */

.slider__view-all {
  display: none;
  text-decoration-line: none;
}

.slider__content {
  display: flex;
  flex-direction: row;
  overflow-x: auto;
  position: relative;
  margin-top: 1em;
}

.slider__content::-webkit-scrollbar {
  display: none;
}

.slider__item {
  margin: 0 5px;
  position: relative;
}

.slider__image {
  border: 4px solid transparent;
  border-radius: 2px;
  width: 40vw;
}

.slider__image:hover {
  border: 4px solid #FFCC00;
  border-radius: 2px;
}

.slider__content-wrapper {
    display: flex;
    flex-direction: column;
}

.slider__item-text {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    height: 5vw;
}

@media screen and (min-width: 600px) {
  .slider {
    height: 60vw;
  }

  .slider__title:after {
    display: none;
  }

  .slider__view-all {
    display: block;
    margin: auto 2rem auto 0;
    color: #fff;
  }

  /* .slider__view-all:after {
    display: inline-block;
    content: ">";
    position: relative;
    margin-left: 4px;
  } */

  .slider__image {
    border: 4px solid transparent;
    border-radius: 2px;
    width: 30vw;
  }
}

@media screen and (min-width: 1024px) {
  .slider {
    height: 35vw;
  }

  .slider__image {
    border: 4px solid transparent;
    border-radius: 2px;
    width: 15vw;
  }
}
