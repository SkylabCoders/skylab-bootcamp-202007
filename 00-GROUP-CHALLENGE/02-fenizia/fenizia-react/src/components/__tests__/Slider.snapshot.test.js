import renderer from "react-test-renderer";
import React from "react";
import Slider from "../Slider";
import { BrowserRouter as Router } from "react-router-dom";

describe("Slider snapshot", () => {
  const tree = renderer.create(
    <Router>
      <Slider />
    </Router>
  );

  it("should match", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
