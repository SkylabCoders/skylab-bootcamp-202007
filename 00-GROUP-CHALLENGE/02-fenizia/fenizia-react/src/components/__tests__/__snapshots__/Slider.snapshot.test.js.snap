// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider snapshot should match 1`] = `
<section
  className="slider"
>
  <div
    className="slider__text"
  >
    <a
      className="slider__title"
      href=""
      onClick={[Function]}
    >
      LIBROS - 
    </a>
    <a
      className="slider__view-all"
      href=""
      onClick={[Function]}
    >
      <p>
        VER TODO
      </p>
    </a>
  </div>
  <div
    className="slider__content"
  />
</section>
`;
